pkgs/games/mindustry/default.nix

```diff
@@ -81,66 +120,76 @@ let
     '';
     outputHashAlgo = "sha256";
     outputHashMode = "recursive";
-    outputHash = "0vzck6hsrvs438s3ikk66qmpak88bmqcb8inqbbjwy7x87d2qsvj";
+    outputHash = "09rwyrg2yv8r499b0dk1bzvymsf98d4j5b95bwd9s4xvrz71is3l";
   };
 
-  # Separate commands for building and installing the server and the client
-  buildClient = ''
+in
+assert stdenv.lib.assertMsg (enableClient || enableServer)
+  "mindustry: at least one of 'enableClient' and 'enableServer' must be true";
+stdenv.mkDerivation rec {
+  inherit pname version unpackPhase;
+
+  prePatch = "cd ../Arc";
+
+  # patches for Arc
+  patches = [
+    ./0001-fix-include-path-for-SDL2-on-linux.patch
+    # upstream fix for https://github.com/Anuken/Arc/issues/40, remove on next release
+    (fetchpatch {
+      url = "https://github.com/Anuken/Arc/commit/b2f3d212c1a88a62f140f5cb04f4c86e61332d1c.patch";
+      sha256 = "1z0vxr906mmi8vk75s3xv6cbn2flv2pikkv5yifmiq1xywhc1gwz";
+    })
+  ];
+
+  postPatch = ''
+    rm natives/natives-desktop/libs/libarc*.so
+    rm backends/backend-sdl/libs/linux64/libsdl-arc*.so
+
+    cd ../Mindustry
+  '' + cleanupMindustrySrc;
+
+  buildInputs = [ SDL2 glew alsaLib ];
+  nativeBuildInputs = [ pkg-config gradle_6 makeWrapper jdk ant copyDesktopItems ];
+  desktopItems = [ desktopItem ];
+
+  buildPhase = with stdenv.lib; preBuild + ''
+    # point to offline repo
+    sed -ie "s#mavenLocal()#mavenLocal(); maven { url '${deps}' }#g" build.gradle
+    sed -ie "s#mavenCentral()#mavenCentral(); maven { url '${deps}' }#g" ../Arc/build.gradle
+  '' + optionalString enableClient ''
+    gradle --offline --no-daemon jnigenBuild -Pbuildversion=${buildVersion}
+    gradle --offline --no-daemon sdlnatives -Pdynamic -Pbuildversion=${buildVersion}
+    patchelf ../Arc/backends/backend-sdl/libs/linux64/libsdl-arc*.so \
+      --add-needed ${glew.out}/lib/libGLEW.so \
+      --add-needed ${SDL2}/lib/libSDL2.so
     gradle --offline --no-daemon desktop:dist -Pbuildversion=${buildVersion}
-  '';
-  buildServer = ''
+  '' + optionalString enableServer ''
     gradle --offline --no-daemon server:dist -Pbuildversion=${buildVersion}
   '';
-  installClient = ''
+
+  installPhase = with stdenv.lib; optionalString enableClient ''
     install -Dm644 desktop/build/libs/Mindustry.jar $out/share/mindustry.jar
     mkdir -p $out/bin
-    makeWrapper ${jre}/bin/java $out/bin/mindustry \
-      ${stdenv.lib.optionalString stdenv.isLinux "--prefix LD_LIBRARY_PATH : ${alsaLib}/lib"} \
+    makeWrapper ${jdk}/bin/java $out/bin/mindustry \
```

Comment:
why jdk (later too)?

Reply:
Maybe this should have been done in a seperate commit with an explanation, but there is a reason:
For modern java versions (>8) there is no JRE package anymore (the JDK is intended to be used for development and at runtime), so the `jre` attribute points to a very outdated version of Java.

Reply:
If there was some specific reason for using `jre` with Java version 8, please unresolve this.